### PR TITLE
[promptflow][BugFix] Print error message when run status is failed

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

If run status is failed, print error message; otherwise, user cannot know why the run is failed in SDK experience.

SDK experience:

image

CLI experience:

image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/operations/_run_operations.py [hunk 1]
```diff
@@ -140,7 +140,10 @@ def stream(self, name: Union[str, Run]) -> Run:
             available_logs = local_storage.logger.get_logs()
             incremental_print(available_logs, printed, file_handler)
             self._print_run_summary(run)
-            # won't print error here, put it in run dict
+            # print error message when run is failed
+            if run.status == RunStatus.FAILED:
+                error_message = local_storage.load_exception()["message"]
+                file_handler.write(f"{error_message}\n")
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/operations/_run_operations.py:146]: Screenshot of this?

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/operations/_run_operations.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/operations/_run_operations.py:146]: added in PR description

**Review by elliotzh — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/operations/_run_operations.py (hunk 1 above)
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/operations/_run_operations.py:146]: maybe we should write this as read in cli?

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/operations/_run_operations.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/operations/_run_operations.py]: updated to use red print for better visibility

**Review by elliotzh — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**